music_assistant/server/controllers/music.py

```diff
@@ -345,6 +345,14 @@ async def add_item_to_favorites(
         """Add an item to the favorites."""
         if isinstance(item, str):
             item = await self.get_item_by_uri(item)
+        # ensure item is added to streaming provider library
+        prov_controller = self.mass.get_provider(item.provider)
+        if (
+            prov_controller
+            and prov_controller.is_streaming_provider
+            and prov_controller.library_edit_supported(item.media_type)
+        ):
+            await prov_controller.library_add(item.item_id, item.media_type)
```

Comment:
Yes, this looks much better!
But... I'm afraid this will fail if the item is already in the library (provider will be 'library' in that case).
So, just to be safe I'd add a check for the provider to have a value, this is also needed for the type checker.

Reply:
Ah I see you do exactly that (I just got distracted by the name prov_controller) - rename that to provider or provider_instance and its all good.